### PR TITLE
feat(PPDSC-2322): add tealiumTrack to instrumentation

### PR DESCRIPTION
**Pull request by Xin00163**
PPDSC-2322

- created `createTealiumTrackHandler` using `window.tealiumTrack()` instead of utag.
- Add tracking to props table in component api

TO DO: 
- Check with Data Analytic team if the implementation is correct and confirm they can receive data from the componentAPI table.
- Also check if we should replace tealiumHandler with tealiumTrackHandler

**I have done:**
- [ ] Written unit tests against changes
- [ ] Written functional tests against the component and/or NewsKit site
- [ ] Updated relevant documentation

**I have tested manually:**
- [x] The feature's functionality is working as expected on Chrome, Firefox, Safari and Edge
- [ ] The screen reader reads and flows through the elements as expected.
- [x] There are no new errors in the browser console coming from this PR.
- [ ] When visual test is not added, it renders correctly on different browsers and mobile viewports (Safari, Firefox, small mobile viewport, tablet)
- [ ] The Playground feature is working as expected

**Before:**

**After:**

**Who should review this PR:**

**How to test:**

### CODE REVIEW

**Comment by newskit-engineering** [comment]
You can preview these changes on:
  Documentation Site: http://ncu-newskit-docs.s3-website-eu-west-1.amazonaws.com/ppdsc-2322-track-props-table
  Storybook: http://ncu-newskit-docs.s3-website-eu-west-1.amazonaws.com/ppdsc-2322-track-props-table/storybook

File: src/instrumentation/handlers/tealium-track.ts [hunk 1]
```diff
@@ -0,0 +1,23 @@
...
+
+export interface ExtendedWindow extends Window {
+  tealiumTrack(e: InstrumentationEvent): void;
+}
+
+const extendedWindow: ExtendedWindow | null =
+  typeof window !== 'undefined' ? ((window as Window) as ExtendedWindow) : null;
+
+export function sendEventTrackingToTealium(e: InstrumentationEvent) {
+  if (extendedWindow) {
```
Comment (mutebg) [inline src/instrumentation/handlers/tealium-track.ts:11]: Maybe we can check for 
```suggestion
  if (extendedWindow && extendedWindow.tealiumTrack) {
```

File: site/components/component-api/component-api.tsx [hunk 2]
```diff
@@ -109,7 +109,13 @@ export const ComponentAPI: React.FC<ComponentAPIProps> = ({components}) => (
...
               <Tabs size="medium">
                 {tabs.map(({label, tabSummary, content}) => (
-                  <Tab label={label} overrides={tabOverrides} key={label}>
+                  <Tab
+                    label={label}
+                    overrides={tabOverrides}
+                    key={label}
+                    eventContext={{component: title, tabName: label}}
+                    eventOriginator="component api"
+                  >
```
Comment (mutebg) [inline site/components/component-api/component-api.tsx:118]: I am not sure if that matters but there are scenarios in which we have multiple API tables on a single page, like Tabs for example we have 2 tables for Tab and Tabs components. Is it worth it to send the page name as well?

Comment (Xin00163) [reply site/components/component-api/component-api.tsx:118]: Will check with Divya about this.

Comment (Xin00163) [reply site/components/component-api/component-api.tsx]: Divya confirmed that we don't need pageName on our side

File: site/pages/components/tooltip.tsx [hunk 3]
```diff
@@ -386,6 +386,7 @@ const TooltipComponent = (layoutProps: LayoutProps) => (
     componentAPI={{
       components: [
         {
+          title: 'Tooltip',
```
Comment (baburay23) [inline site/pages/components/tooltip.tsx:389]: have these been added/updated so that the components are tracked correctly?

Comment (Xin00163) [reply site/pages/components/tooltip.tsx:389]: Yes exactly. I've made title prop mandatory now

Comment (mutebg) [reply site/pages/components/tooltip.tsx:389]: That changes a bit the docs website:
from this:
before

to this:
after

Have you discussed that with the design team?

Comment (Xin00163) [reply site/pages/components/tooltip.tsx:389]: Hey I thought this is a mistake as other pages are using component name as title but will let design know

**Review by mutebg — APPROVED** [review]

**Review by baburay23 — DISMISSED** [review]

**Review by baburay23 — APPROVED** [review]

**Review by mutebg — APPROVED** [review]

**Pull request merged**